### PR TITLE
Fix for https://github.com/mongoid/moped/issues/109

### PR DESCRIPTION
**Pull request by EmilyMcCarthy**

### CODE REVIEW

File: lib/moped/protocol/reply.rb [hunk 1]
```diff
@@ -108,7 +108,12 @@ def query_failed?
       # @since 1.2.10
       def unauthorized?
         result = documents[0]
-        result["code"] == UNAUTHORIZED || result["assertionCode"] == UNAUTHORIZED
+        return false if result.nil?
+        
+        err = error_message(result)
+        UNAUTHORIZED.include?(result["code"]) ||
```
Comment (keck) [inline lib/moped/protocol/reply.rb:114]: can I get an explicit 'return ' at the beginning of this phrase?

Comment (keck) [inline lib/moped/protocol/reply.rb:114]: For clarity I might even prefer fall-thru style, but this is purely opinion.

```
  return true if UNAUTHORIZED.include?(result["code"])
  return true if UNAUTHORIZED.include?(result["assertionCode"])
  return true if (err && (err =~ /unauthorized/ || err =~ /not authorized/))
  return false
```

Comment (EmilyMcCarthy) [inline lib/moped/protocol/reply.rb:114]: on one hand, I agree. On the other, I c/p'd from
https://github.com/mongoid/moped/commit/0416e5d9db552ebef299ed4748e7541cf38710fa
and kinda want to keep it the same. I dunno, thoughts?

**Comment by keck** [comment]
LGTM

File: lib/moped/protocol/reply.rb (hunk 1 above)
Comment (keck) [inline lib/moped/protocol/reply.rb:114]: Maybe just add that url in a comment or call out the original upstream file + function it was copy pasta'd from for reference, at most.  I don't think it's undecipherable as-is.

**Pull request merged**